cpp/src/io/parquet/page_data.cu

```diff
@@ -1690,7 +1728,13 @@ extern "C" __global__ void __launch_bounds__(block_size)
           switch (dtype) {
             case INT32: gpuOutputFast(s, src_pos, static_cast<uint32_t *>(dst)); break;
             case INT64: gpuOutputFast(s, src_pos, static_cast<uint2 *>(dst)); break;
-            default: gpuOutputDecimalAsFloat(s, src_pos, static_cast<double *>(dst), dtype); break;
+            default:
+              if (s->col.strict_decimal_type) {
```

Comment:
Why would we only want to do this with strict decimal types? That flag means that we blow up if we have to coerce the value into a double. I think we would want this all the time and read decimals as decimals whenever possible.

Reply:
Yes, you are right. I misunderstood the meaning of `strict_decimal_types`. Thanks for correcting me.